sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSQLParser.scala

```diff
@@ -31,7 +31,12 @@ import org.apache.spark.sql.types.StringType
  *
  * @param fallback A function that parses an input string to a logical plan
  */
-class SparkSQLParser(fallback: String => LogicalPlan) extends AbstractSparkSQLParser {
+class SparkSQLParser(fallback: => ParserDialect) extends AbstractSparkSQLParser {
```

Comment:
I thought the same thing. The problem is that we we pass the ```SQLContext.getDialect()``` method to the parser (same as before): this allows us to change the dialect whenever we want.